projects/extension/src/containers/Popup.tsx

```diff
@@ -26,71 +20,63 @@ interface ChainDetails {
 
 const Popup: FunctionComponent = () => {
   const [connChains, setConnChains] = useState<PopupChain[] | undefined>()
-
-  const [bg, setBg] = useState<Background | undefined>()
   const [showModal, setShowModal] = useState<boolean>(false)
 
-  useEffect(() => {
-    chrome.runtime.getBackgroundPage((backgroundPage) => {
-      setBg(backgroundPage as Background)
-    })
-  }, [])
-
-  const refresh = useCallback(() => {
-    if (!bg) return
-    const allChains: PopupChain[] = []
-
-    bg.uiInterface.chains.forEach((c) => {
-      const i = allChains.findIndex((i) => i.chainName === c.chainName)
-      const { peers, isSyncing, chainId, bestBlockHeight } = c
-      if (i === -1) {
-        allChains.push({
-          chainName: c.chainName,
-          details: [
-            {
-              tabId: c.tab?.id,
-              url: c.tab?.url,
+  const refresh = () => {
+    environment.get({ type: "activeChains" }).then((chains) => {
+      const allChains: PopupChain[] = []
+      ;(chains || []).forEach((c) => {
+        const i = allChains.findIndex((i) => i.chainName === c.chainName)
+        const { peers, isSyncing, chainId, bestBlockHeight } = c
```

Comment:
```suggestion
        const {
          peers,
          isSyncing,
          chainId,
          bestBlockHeight,
          chainName,
          tab: { id, url },
        } = c
        const i = allChains.findIndex((i) => i.chainName === chainName)
```

Reply:
That's code that you wrote, I just changed the indentation